### ISSUE TITLE
Adding waypoint to existing route misses supercharging opportunity

### ISSUE THREAD

**Issue by EVGrokker** [aesculus/EVTO-App-Feedback#492]
v 1.2.0 (62)

>Car Data: S85, Charger: 48, Wheels: 19, Tires: 0, Pano: 0, 5 Seat: 0, Rear CC: 0, Bat Life: 100, Payload: 100, PF: 0, SA: 0, Temp: 24 
Device Info: iPhone; CPU iPhone OS 10_3_3 like Mac OS X 
Settings: Reserve SoC:15, Default SoC: 90, Region: 0, Units: 0 
Version: 1.2.0 (62) 
IAP: Pro:1, Sub:1 
Location: Unknown 
Trip: Grants Pass Supercharger, [ap8jy9qe6l707qd3ycdzwyl]

This trip was initially constructed from Seattle to Santa Barbara, then segmented. I then added a Bend waypoint in the first segment. The arrival SoC for the first leg is -7%, even though there is a supercharger in Bend available before or after the waypoint.

I tried manually optimizing but it still wouldn't add it.

**Comment by aesculus** [comment]
Ouch. OK, on it. Like a dog chasing it's tail.

**Comment by aesculus** [comment]
This may take some time. My favorite algorithm has a flaw in it.  Only took me two weeks to unbundle it the last time. :-)

At least now I know where the issue is.  It might be trivial or maybe not.

**Comment by aesculus** [comment]
Can you try this again now?  I think I found (the) a mistake. It now works on my tests of the same trip.

Basically I had changed my model for starting at 10% instead of 15% and I failed to realize that I still had the older algorithms in place for certain charging stops. So a 90% departure could not be realized by a 95% model.

We will have to keep an eye on this to make sure other combinations don't cause issues.

BTW I noticed that I am using the settings min soc to determine color on SoC chart vs what the incident has set. With my settings Min SoC set to 20% the last bit of the leg to Grants Pass came in yellow even though it was within the acceptable SoC of the waypoint of 15% arrival.

**Comment by EVGrokker** [comment]
v 1.2.0 (64)

Following the same steps, adding a waypoint in Bend, then manually optimizing, now correctly adds the Bend supercharger en route to the destination.

**Comment by EVGrokker** [comment]
v 1.2.0 (64)

Trip: Oxford Suites Portland - Jantzen Beach, [ylz3an9psfj2zgeb7ehhnr]
>Car Data: X90D, Charger: 48, Wheels: 20, Tires: 0, Pano: 0, 5 Seat: 0, Rear CC: 0, Bat Life: 100, Payload: 156, PF: 0, SA: 0, Temp: 21 
Device Info: iPhone; CPU iPhone OS 10_3_3 like Mac OS X 
Settings: Reserve SoC:20, Default SoC: 90, Region: 0, Units: 0 
Version: 1.2.0 (64) 
IAP: Pro:1, Sub:1 
Location: Unknown 

Not working for me. I created a trip from Seattle to Oxford Suites in Portland. With the car metrics described above, the arrival SoC was 18%, no charging stops en route.

I then added a segment to Bend, but received an error that no superchargers were available. However, there are two in the Portland area, and the Centralia supercharger available before Portland. The Trip Browser does not show the second segment, even though toggling to the Main screen and back to the Plan screen does show it. 

Manually optimizing does not find the available supercharging either.

I reopened this because of #274.

**Comment by aesculus** [comment]
Try this:  Go to Edit Segment Details and select Save. Then optimize.

**Comment by aesculus** [comment]
>I then added a segment to Bend, but received an error that no superchargers were available. However, there are two in the Portland area, and the Centralia supercharger available before Portland. The Trip Browser does not show the second segment, even though toggling to the Main screen and back to the Plan screen does show it.

I did not have this experience at all with 64. I did the above exactly and came up with the route with Centrallia and Sandy, OR as the two charging stops.

Did you do the Auto selection when you added the segment? I did.

**Comment by aesculus** [comment]
Interestingly I did not have my settings Min Arrival SoC set to 20% but 15%. When I did that and created a new trip it routed differently. This time using Woodburn and Detroit Lake SC to the south.  I wonder if the weather changed during the time or just having the Min SoC differently caused this.  Both were valid routes. 

I did not check out the trip times to see which was better. Sometimes it's only a few minutes.

**Comment by EVGrokker** [comment]
v 1.2.0 (65)

I tried recreating this trip from scratch, starting with Seattle - Oxford Suites in Portland. I mistakenly chose Edit Segment Details from the menu, then saved. The update progress hangs at about 80%, even though I didn't change anything when I opened that dialog.

I cancelled, then tried adding Bend as a second segment, and the update dialog again hangs.

Optimize Routing does nothing.

Trip: Oxford Suites Portland - Jantzen Beach, [6t0etyxtg5uodrfdsh86wj]

>Car Data: X90D, Charger: 48, Wheels: 20, Tires: 0, Pano: 0, 5 Seat: 0, Rear CC: 0, Bat Life: 100, Payload: 156, PF: 0, SA: 0, Temp: 21 
Device Info: iPhone; CPU iPhone OS 10_3_3 like Mac OS X 
Settings: Reserve SoC:20, Default SoC: 90, Region: 0, Units: 0 
Version: 1.2.0 (65) 
IAP: Pro:1, Sub:1 
Location: Unknown

**Comment by aesculus** [comment]
> Optimize Routing does nothing.

Bug. Try again on 66.

**Comment by aesculus** [comment]
I confirmed this works OK on the iPad. Did not try the iPhone but doubt it would make any difference.

I did discover a "issue" which we should discuss.  

When I optimized with the settings of 15% arrival SoC that is what the destination was set at.  It got a yellow pushpin because you had set the destination at 20%. If I saved the destination then it upped the charge at Sandy to make sure that the arrival was 20%.

So the question is should I have automatically used the destination value vs the settings value? I think yes, and I am kind of wondering why it did not.
